File: src/main/java/com/opencode/pycharm/actions/InlinePromptAction.java
package com.opencode.pycharm.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.CommonDataKeys;
import com.intellij.openapi.ui.Messages;
import com.opencode.pycharm.services.OpenCodeStateService;

public class InlinePromptAction extends AnAction {
    @Override
    public void actionPerformed(AnActionEvent e) {
        var project = e.getProject();
        var editor = e.getData(CommonDataKeys.EDITOR);
        if (project == null || editor == null) {
            return;
        }

        String selection = editor.getSelectionModel().getSelectedText();
        if (selection == null || selection.isBlank()) {
            Messages.showInfoMessage(project, "Please select code first.", "OpenCode");
            return;
        }

        var state = OpenCodeStateService.getInstance(project);
        var conversation = state.getConversations().isEmpty()
                ? state.createConversation("Inline Prompt")
                : state.getConversations().get(0);

        state.appendMessage(conversation.id(), "user", "Please explain/refactor this selection:\n" + selection);
        state.appendMessage(
                conversation.id(),
                "assistant",
                "(Mock) Inline assistant received " + selection.length() + " chars. Connect here to OpenCode API streaming response."
        );

        Messages.showInfoMessage(project, "Selection sent to OpenCode chat.", "OpenCode");
    }
}

File: src/main/kotlin/com/opencode/pycharm/actions/InlinePromptAction.kt
package com.opencode.pycharm.actions

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.editor.EditorFactory
import com.intellij.openapi.project.DumbAware
import com.intellij.openapi.ui.popup.JBPopupFactory
import com.intellij.ui.components.JBTextField
import com.opencode.pycharm.service.OpenCodeService
import java.awt.BorderLayout
import javax.swing.JButton
import javax.swing.JLabel
import javax.swing.JPanel

class InlinePromptAction : AnAction(), DumbAware {
    override fun actionPerformed(e: AnActionEvent) {
        val project = e.project ?: return
        val editor = e.getData(com.intellij.openapi.actionSystem.CommonDataKeys.EDITOR)
            ?: EditorFactory.getInstance().allEditors.firstOrNull { it.project == project }
            ?: return

        val promptField = JBTextField()
        val panel = JPanel(BorderLayout(8, 8))
        panel.add(JLabel("OpenCode 指令（Cmd/Ctrl+Enter 接受，Esc 拒绝）"), BorderLayout.NORTH)
        panel.add(promptField, BorderLayout.CENTER)
        val submit = JButton("发送")
        panel.add(submit, BorderLayout.EAST)

        val popup = JBPopupFactory.getInstance()
            .createComponentPopupBuilder(panel, promptField)
            .setTitle("Inline Edit")
            .setRequestFocus(true)
            .setResizable(false)
            .setMovable(true)
            .createPopup()

        submit.addActionListener {
            val prompt = promptField.text.trim()
            if (prompt.isNotEmpty()) {
                OpenCodeService.getInstance().requestInlineEdit(prompt, editor)
                OpenCodeService.getInstance().sendStateSync(editor)
            }
            popup.cancel()
        }

        popup.showInBestPositionFor(editor)
    }
}
